benchmarks/ALCE/ELI5/eli5.py

```diff
@@ -0,0 +1,152 @@
+import json
+import os
+import time
+
+from datasets import Dataset
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+import rageval as rl
+from rageval.models.openai import OpenAILLM
+from utils import make_demo, create_eli5_eval_dataset
+
+
+class ELI5:
+
+    def __init__(self, args):
```

Comment:
There are several questions:
 1. Does it necessary to implement the benchmark task as a `Class`? we can learn more from the [BIG-Bench](https://github.com/google/BIG-bench/tree/main/bigbench/benchmark_tasks) repo from google.
 2. The initialization function is too complicated. Most of the code in this function are not necessary, when I only need to evaluate some existing result file.

Reply:
> There are several questions:
> 
> 1. Does it necessary to implement the benchmark task as a `Class`? we can learn more from the [BIG-Bench](https://github.com/google/BIG-bench/tree/main/bigbench/benchmark_tasks) repo from google.
> 2. The initialization function is too complicated. Most of the code in this function are not necessary, when I only need to evaluate some existing result file.

1. The reason for implementing the benchmark as a class is that the differences between benchmarks are very large. We may need to implement more complex benchmarks later, which may involve retrieval, even involve rewrite and rerank, rather than a single generation task.
2. Now `__init__` has been simplified, all initialization of the model and data set has been migrated to the `generate_init` function. The parameter interface of the `evaluate` function has been modified, the file address can be passed in directly now.
  As follows
  ```
  eli5 = ELI5(args)
  gen_result = eli5.generate()
  gen_result_path = eli5.save_result(gen_result)
  # gen_result_path = ".rageval/results/eli5-bm25-Llama-2-7b-chat-hf-vanilla-shot2-ndoc5.json"
  eval_result = eli5.evaluate(gen_result_path)
  ```

Reply:
Maybe the pyseudo code shoult be like this:

```
eli5 = ELI5(args)

# in case we need to predict the result with some pre-defined model.
model = OpenAILLM()
eli5.init_model(model)
_ = eli5.predict()   # here we can also use "result = eli5.predict()" if we need the result
eli5.evaluate() # there should be two version of  evaluate func, one is `evaluate()` without params, the other is `evaluate(path='./XX/XX/XX.json')`
eli5.save_result(path='./XX/XX/XX.json')

# in other case where we direct evaluate some existing result
eli5.evaluate(path='./XX/XX/XX.json)
eli5.save_result(path='./XX/XX/')
```